atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/memory/InMemoryAsyncAtlasDbFactory.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.memory;
+
+import java.util.Optional;
+import java.util.function.LongSupplier;
+import java.util.function.Supplier;
+
+import com.google.auto.service.AutoService;
+import com.palantir.atlasdb.config.LeaderConfig;
+import com.palantir.atlasdb.keyvalue.api.KeyValueService;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.keyvalue.impl.AsyncInitializeableInMemoryTimestampService;
+import com.palantir.atlasdb.keyvalue.impl.InMemoryKeyValueService;
+import com.palantir.atlasdb.qos.QosClient;
+import com.palantir.atlasdb.spi.AtlasDbFactory;
+import com.palantir.atlasdb.spi.KeyValueServiceConfig;
+import com.palantir.atlasdb.spi.KeyValueServiceRuntimeConfig;
+import com.palantir.atlasdb.util.MetricsManager;
+import com.palantir.atlasdb.versions.AtlasDbVersion;
+import com.palantir.timestamp.InMemoryTimestampService;
+import com.palantir.timestamp.TimestampService;
+
+@AutoService(AtlasDbFactory.class)
+public class InMemoryAsyncAtlasDbFactory implements AtlasDbFactory {
```

Comment:
this should also exist only in test code! It's not valuable for a consumer of Atlas to have this code around